### PR TITLE
Update worker.js to work with MODULARIZE

### PR DESCRIPTION
**Pull request by tlively**
Recent work on proxying had broken MODULARIZE with pthreads due to lack of test
coverage. Fix the breakage and add a test exercising the proxying API with
MODULARIZE.

### CODE REVIEW

**Comment by tlively** [comment]
Only the last commit is relevant: https://github.com/emscripten-core/emscripten/pull/16782/commits/cca8dbfd80d423706bcdc2ed8c9d70553df522de

I'll rebase and force-push this once all the previous PRs have been merged.

**Comment by tlively** [comment]
Hmm, so I took a look at the asan test faillure and it looks like `urlOrBlob` on the load message for the initial workers is the expected string used to load the main script, but then later in the test some additional workers are spawned and `urlOrBlob` is undefined, causing the worker to go down the code path that calls `URL.createObjectURL`. I'm pretty sure `URL.createObjectURL` is a red herring; the real issue is that `urlOrBlob` should not be undefined. I wasn't able to figure out _why_ it was undefined, though. @sbc100, would you want to take a stab at this?

File: tests/test_core.py [hunk 1]
```diff
@@ -2604,12 +2604,22 @@ def test_pthread_equal(self):
...
+  def test_pthread_proxying(self, modularize):
     self.set_setting('EXIT_RUNTIME')
     self.set_setting('PROXY_TO_PTHREAD')
     self.set_setting('INITIAL_MEMORY=32mb')
+    args = []
+    if modularize:
+      self.set_setting('MODULARIZE')
+      self.set_setting('EXPORT_NAME=ModuleFactory')
+      create_file('extern-post.js', 'ModuleFactory();')
+      args = ['--extern-post-js=extern-post.js']
```
Comment (sbc100) [inline tests/test_core.py:2620]: This code will run on all threads.. is that needed/good?

**Review by sbc100 — APPROVED** [review]

File: tests/test_core.py (hunk 1 above)
Comment (tlively) [reply tests/test_core.py:2620]: Oh, good point. That's not what I intended, and I wonder if it is the cause of the problems I'm seeing.

**Pull request merged**